### ISSUE TITLE
high system load after upgrade to 8.2.1

### ISSUE THREAD

**Issue by torsten-winkler** [owncloud/core#20604]
After upgrading from 8.2.0 to 8.2.1 our server has very high system load, there are many apache processes and sometimes mysql and php as well. The system load is near 100% at all times, and the web interface is very unresponsive. 

The upgrade itself finished OK, I just got the usual warnings that some apps had been disabled in the process. After that I re-enabled the apps without problems. None of the apps needed to be updated. A few minutes after that I noticed that the server was slowing down.

See example output from top below:

PID USER      PR  NI    VIRT    RES    SHR S  %CPU %MEM     TIME+ COMMAND                                                                                                                                        
1253 mysql     20   0 2849936 230904   7600 S  88.4  1.4 958:27.21 mysqld                                                                                                                                         
8894 www-data  20   0  398784  33720  20640 R  85.5  0.2   0:28.72 apache2                                                                                                                                        
8948 www-data  20   0  396608  27040  16544 R  75.2  0.2   0:36.40 apache2                                                                                                                                        
8955 www-data  20   0  397360  27592  16332 R  53.2  0.2   0:06.60 apache2                                                                                                                                        
8971 www-data  20   0  393444  23732  16164 S  46.4  0.1   0:01.45 apache2                                                                                                                                        
8874 www-data  20   0  404804  41064  21732 S  22.1  0.3   1:04.32 apache2                                                                                                                                        
8969 www-data  20   0  393444  23268  15716 S  19.5  0.1   0:01.18 apache2                                                                                                                                        
8952 www-data  20   0  399304  35116  21320 S  10.2  0.2   0:21.51 apache2                                                                                                                                        

I tried disabling all non-standard apps. The web interface is still unresponsive. It takes approx. 10 secs to load "files" page. Whenever I try to load/reload a page, several apache processes appear, each one using up 50-100% CPU. This looks seriously messed up. I am considering going back to 8.2.0 and restoring the sql database to a backup from yesterday. Any ideas what causes this, and how to fix this would be extremely appreciated.

ownCloud version: 8.2.1
approx. 40 active users
Webserver: Apache
Database: MySQL (database dump size: approx. 800 MB)
OS: Linux / Ubuntu 14.04.3 LTS
PHP version: 5.5.*

**Comment by MorrisJobke** [comment]
If I interpret the output correct: It's mostly MySQL that consumes all the CPU time. Or am I wrong?

**Comment by torsten-winkler** [comment]
-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA256

Dear Morris,

thanks for the quick reply.

At the moment there are mostly apache2 processes, each using between
20 and 100 % CPU. mysqld pops up approx. every 10-15 minutes, using up
to 250% CPU (on a quadcore machine) but the process vanishes after a
couple of minutes. I have also seen a php process, they are even more
rare, maybe once an hour.

What bothers me most is the unresponsive interface. The number of
high-CPU apache2 processes immediately rises after I load/reload a page.

Best regards,

Torsten


Dr. Torsten Winkler
Otto-Diels-Institut für Organische Chemie
Otto-Hahn-Platz 4
R. 216
24118 Kiel

On 19.11.2015 11:11, Morris Jobke wrote:
> If I interpret the output correct: It's mostly MySQL that consumes
> all the CPU time. Or am I wrong?
> 
> — Reply to this email directly or view it on GitHub 
> <https://github.com/owncloud/core/issues/20604#issuecomment-158012482>
.
>
-----BEGIN PGP SIGNATURE-----
Version: GnuPG v2

iQEcBAEBCAAGBQJWTaG5AAoJECZuWkFApgJwfqgH/2/u6eiDpObvIntvwiB6hqID
bws/lk82vWSNB09cL8p7QseApkB3N8SsaXl30khL8SnfDpWLpAPl72Ie+jLrqnCr
z2Z2xaiIJDPsO+FhWehrPOnWG7nWiZJpR5ISua5vGGYB5G9qL+fO3p34fIGQ9GoT
q7tMTOa3HoXuAT5iLS75y6XxS+zYV77f145ueyweVFRmByUXVVXsvrE6/1K8dbrQ
bPrZhavJGnzUoeN2KEa9FGyGev3bPKNJgh103c7kIIAC2plkW6JwQ0J89kwiIFT3
ic+t+oxLvDei69R0OEkPMumaZrR/5TFEEkc4uIrsnzflJfigbGE5pdhuiFcP4ZA=
=U06Q
-----END PGP SIGNATURE-----

**Comment by felixboehm** [comment]
What are your cronjobs doing, running every 10 min.? Please look into table oc_jobs

How big are your tables oc_activity, oc_files_trash, oc_files_versions, oc_users and oc_filecache
Any very large?

**Comment by torsten-winkler** [comment]
Again, thanks for your reply. OC cron is active with default settings, mainly to run updates for the news app. But the news app is disabled now, to no effect.

The largest tables (data_length) in the owncloud database are:
oc_activity: 250871808
oc_filecache: 320339968
oc_news_items: 2405433344
oc_oc_news_items_0a8313e8cb7a3: 929038336
oc_oc_news_items_75e978aff8cc1: 1251999744
oc_contacts_cards: 23183360

Does that mean that the news app is (or was) somehow out of control? Is it possible to just drop / delete the oc_news* tables to see if speed increases?

**Comment by MorrisJobke** [comment]
> oc_filecache: 320339968

Oh....320 million file cache entries is quite a lot. Every user is having 10 million files? How are they distributed? All in subdirectories or maybe there is one directory with a lot of files ... that can somehow affect the loading time of a directory listing and then influence other requests too.

**Comment by MorrisJobke** [comment]
Something like `SELECT storage, COUNT(*) c FROM oc_filecache GROUP BY storage ORDER BY c;` might be interesting.

**Comment by MorrisJobke** [comment]
And then check the corresponding oc_storages entries for the top results in this query.

**Comment by xn--nding-jua** [comment]
I think he means "Byte" with data-length: oc_filecache: 320339968 = 305,5MB which is more realistic, isn't it?

**Comment by torsten-winkler** [comment]
yes, I think it means bytes. I refer to the "data_length" column of
 show table status from owncloud;

This is the output of:
 
mysql> SELECT storage, COUNT(*) c FROM oc_filecache GROUP BY storage ORDER BY c;
+---------+--------+
| storage | c      |
+---------+--------+
|      70 |      2 |
|      72 |      2 |
|      78 |      2 |
|      79 |      2 |
|      80 |      2 |
|      58 |      3 |
|      59 |      3 |
|      76 |      4 |
|      61 |      6 |
|      43 |      7 |
|      64 |     12 |
|      84 |     12 |
|      66 |     12 |
|      67 |     12 |
|      69 |     19 |
|      82 |     22 |
|      55 |     25 |
|      83 |     28 |
|      49 |     31 |
|      68 |     51 |
|      57 |     67 |
|      45 |     92 |
|      63 |    618 |
|      60 |    792 |
|      39 |    856 |
|      47 |   1250 |
|      81 |   1905 |
|      42 |   2274 |
|      74 |   3546 |
|      56 |   4613 |
|      50 |   5241 |
|      31 |   6659 |
|      53 |   9595 |
|      37 |   9610 |
|      40 |  10289 |
|      34 |  14064 |
|      35 |  15631 |
|      46 |  18102 |
|      62 |  19038 |
|      54 |  23683 |
|      32 |  28031 |
|      48 |  34392 |
|      44 |  41562 |
|      75 |  44554 |
|      36 |  59260 |
|      33 |  69837 |
|      52 | 118826 |
|      38 | 152858 |
|      41 | 364998 |
+---------+--------+
49 rows in set (18.80 sec)

**Comment by MorrisJobke** [comment]
> I think he means "Byte" with data-length: oc_filecache: 320339968 = 305,5MB which is more realistic, isn't it? Noone has 23183360/40=579584 contacts in his addressbook :-D

Yes. This makes more sense.

**Comment by MorrisJobke** [comment]
cc @BernhardPosselt regarding news app performance impacts

**Comment by BernhardPosselt** [comment]
I dont understand why these two tables are listed, indices?
* oc_oc_news_items_0a8313e8cb7a3
* oc_oc_news_items_75e978aff8cc1

**Comment by torsten-winkler** [comment]
first of all: Thank you very much for tying to help me.

I don't understand the various oc_news_indices tables either. But I'm no expert in mysql and programming in general. I see two imaginable reasons:
Option A) 
The news app has split its table into several parts because the size has become too large with time
Option B) 
The news aggregator job has failed / crashed at least twice, resulting in orphaned oc_news tables

As it stands, I will probably not re-enable the news app anyway for the time being. Will it harm the system if I drop all news-related tables from my database? Is there any chance that speed will increase again?

The reason I'm asking: Our server has 40 active users (who are already bugging me about the problems). I can either drop the news tables or just restore a database dump from before the 8.2.1 upgrade and go back to 8.2.0 for the time being

**Comment by tflidd** [comment]
I can confirm that `news_items` can take up to 200-300 MB per user when used extensively.

**Comment by BernhardPosselt** [comment]
You can drop all the tables, see https://github.com/owncloud/news#how-do-i-reset-the-news-app however I'd ask your users to export their stuff before deleting everything

As for performance: It depends on the system but the update process is ofc very I/O intensive. It's recommended to use a separate install for bigger systems that only the feed updates using the also provided feed updater.

However I doubt that this will fix your issue since a core update apparently triggered the issue. I'd just disable the News app (no tables are dropped) and check again

**Comment by torsten-winkler** [comment]
The news app has been disabled for 6 hours, which didn't reduce the load. I just dropped the oc_news* tables, which also had mo effect. As the machine already starts beeping probably due to overheating, I just put the server into maintenance mode and shut off apache2. I think I will indeed do the rollback to a clean 8.2.0 and mysql restore. If that does not help either, then I'm completely scr**ed...

**Comment by MorrisJobke** [comment]
The problem here is: "What is actually causing this high load?"

@torsten-winkler Do you have more precise logs? A good start would be to log the request time to the apache access log and the analyze this then.

http://www.moeding.net/archives/33-Logging-Apache-response-times.html

Then we can look if there are specific requests that took long.

As a first try you can also use the web developers console in your browser (go to the network tab) to see which requests took how long.

**Comment by torsten-winkler** [comment]
I just tried to activate the custom log, but somehow it doesn't log anything. 

I attached the output of Google Chrome's Network log. owncloud_files.har.txt was captured loading the files overview, owncloud_calendar.har.txt refers to the calendar page.

[owncloud_calendar.har.txt](https://github.com/owncloud/core/files/39017/owncloud_calendar.har.txt)
[owncloud_files.har.txt](https://github.com/owncloud/core/files/39018/owncloud_files.har.txt)

**Comment by torsten-winkler** [comment]
I just had to reboot the owncloud server due to power supply issues (this is really not my kind of day...). I was ready to reset the database, but out of curiosity, I turned maintenance mode off. So far, there has been no high CPU load any more. Perhaps the mysql restart, now without the dropped large oc_news tables, might be responsible. 

This is what top looks like at the moment:

 PID USER      PR  NI    VIRT    RES    SHR S  %CPU %MEM     TIME+ COMMAND                                                                                                                                        
1967 www-data  20   0  397940  29496  17380 S   1.7  0.2   0:00.94 apache2                                                                                                                                        
1984 www-data  20   0  396348  25748  15168 S   1.7  0.2   0:00.19 apache2                                                                                                                                        
1985 www-data  20   0  396588  28556  17704 S   1.7  0.2   0:00.52 apache2                                                                                                                                        
1275 mysql     20   0 1807204 131676   7476 S   0.9  0.8   0:11.52 mysqld                                                                                                                                         

Below you find another two log files, recorded after the reboot. I would be extremely glad if the server continues to run smoothly. I will run only the most necessary non-standard apps for the time being (contacts, calendar, tasks).

[owncloud_calendar_after_reboot.har.txt](https://github.com/owncloud/core/files/39103/owncloud_calendar_after_reboot.har.txt)
[owncloud_files_after_reboot.har.txt](https://github.com/owncloud/core/files/39104/owncloud_files_after_reboot.har.txt)

**Comment by MorrisJobke** [comment]
I didn't find an easy way to inspect the har files. Can you simply post a screenshot of the web dev console? ;)

**Comment by torsten-winkler** [comment]
...no problem, here are some screenshots
![screenshot_calendar_bottom](https://cloud.githubusercontent.com/assets/15922846/11276975/a4a58134-8ee4-11e5-8d12-e32284aeddf8.png)
![screenshot_calendar_top](https://cloud.githubusercontent.com/assets/15922846/11276976/a4a5dc88-8ee4-11e5-98e9-d2bfb228ed72.png)
![screenshot_files_bottom](https://cloud.githubusercontent.com/assets/15922846/11276978/a4ac0b12-8ee4-11e5-8fcc-7d5592d0fc27.png)
![screenshot_files_top](https://cloud.githubusercontent.com/assets/15922846/11276977/a4a9dba8-8ee4-11e5-9b3d-f94d6243b74a.png)

**Comment by torsten-winkler** [comment]
...of course, these screenshots were all taken after the reboot / mysql restart, so this ought to be the working state. Probably (hopefully?!) I will not be able to provide screens from the unresponsive pages again.

**Comment by MegaV0lt** [comment]
Can it be something with th eFile Locking that has changed: http://j.mp/1P0pXv0

**Comment by MorrisJobke** [comment]
> Can it be something with the File Locking that has changed in 8.2: http://j.mp/1P0pXv0

Could be but doesn't explain, why it is gone after the reboot.

**Comment by torsten-winkler** [comment]
If file locking had already changed with 8.2.0, then the problem should have occurred two weeks earlier when I upgraded from the latest 7.* to 8.2.0. I can only imagine the problem being mysql-related as well as connected to the news app and/or the large database size.I restarted apache several times, to no effect. I even rebooted the machine (with news app still active), still the problem persisted. The latest reboot caused a mysql restart after dropping several tables, which finally seems to do the trick. So far, the machine is still fast and responsive
